geo/src/algorithm/closest_point.rs

```diff
@@ -49,7 +49,7 @@ impl<F: Float> ClosestPoint<F> for Point<F> {
 }
 
 #[allow(clippy::many_single_char_names)]
-impl<F: Float + HasKernel> ClosestPoint<F> for Line<F> {
+impl<F: Float> ClosestPoint<F> for Line<F> {
```

Comment:
It's possible that people will assume this Float refers to num_traits::Float. I could easily be convinced to call it something more explicit like `GeoFloat`.

Reply:
May be call it `geo::Float`, but `use geo::Float as GeoFloat` to clarify?  Don't have any hard opinions here either.

Reply:
I vote for GeoFloat, because it's unambiguous. Using Float will lead to people assuming that they know what it is, whereas with GeoFloat they'll have to look it up and read the docstring.

Reply:
Ok, I've renamed geo::Float -> geo::GeoFloat and added some doc comments.